### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by pganssle**
Adds a pull request template to make it clear what should be in most PRs.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<!-- First time contributors: Take a moment to review CONTRIBUTING.md! -->
+<!-- Remove sections if not applicable -->
+## Summary of changes
+
+<!-- Summary goes here -->
+
+Closes <!-- issue number here -->
```
Comment (jbrockmendel) [inline .github/pull_request_template.md]: I'd make this a checkbox in the checklist section.  Fine either way.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+### Pull Request Checklist
```
Comment (jbrockmendel) [inline .github/pull_request_template.md]: Possible include a style check on the diff.  From the pandas template:

```
[ ] passes git diff upstream/master -u -- "*.py" | flake8 --diff
```

Comment (pganssle) [reply .github/pull_request_template.md]: For anything automated like that I'd rather just add it to the CI.

File: .github/pull_request_template.md (hunk 1 above)
Comment (pganssle) [reply .github/pull_request_template.md]: Yeah, I debated that. The reason I decided against it were:  1. that it's not really an "action item", it's a piece of information to convey in the PR and 2. I wasn't sure that the github auto-closing checker would recognize it in a checklist.

Comment (pganssle) [reply .github/pull_request_template.md]: I think the latter is probably not a real concern, but I'll leave it as is unless it's confusing people.

Comment (jbrockmendel) [reply .github/pull_request_template.md]: Fair enough, not a strong preference.  FWIW the pandas template puts the "closes" in the checklist.

File: .github/pull_request_template.md (hunk 2 above)
Comment (jbrockmendel) [reply .github/pull_request_template.md]: Sure.  I'm a happy camper

**Pull request merged**